crates/noirc_evaluator/src/ssa/acir_gen/acir_mem.rs

```diff
@@ -1,22 +1,154 @@
-use crate::ssa::{
-    acir_gen::InternalVar,
-    context::SsaContext,
-    mem::{ArrayId, MemArray},
+use crate::{
+    ssa::{
+        acir_gen::InternalVar,
+        context::SsaContext,
+        mem::{ArrayId, MemArray},
+    },
+    Evaluator,
 };
-use acvm::acir::native_types::Witness;
+use acvm::{
+    acir::{
+        circuit::{directives::Directive, opcodes::Opcode as AcirOpcode},
+        native_types::{Expression, Witness},
+    },
+    FieldElement,
+};
+
 use iter_extended::vecmap;
 use std::collections::BTreeMap;
 
+use super::{
+    constraints::{self, mul_with_witness, subtract},
+    expression_from_witness,
+    operations::{self},
+};
+
+/// Represent a memory operation on the ArrayHeap, at the specified index
+/// Operation is one for a store and 0 for a load
```

Comment:
Why are we representing stores as `1` and loads as `0` rather than a dedicated enum?

Reply:
Because we need to have FieldElements.

Reply:
Why do we need to represent the operation via field elements?

Reply:
This is the purpose of noir, to represent the program as field elements.

Reply:
Can't we just convert to an `Expression` within `acir_gen`? It doesn't seem like we'd ever pass a non-hardcoded `Expression` into `push`.

Reply:
That does not answer my question. Why specifically do they need to be represented as 1 and 0 instead of e.g. different gates?

Reply:
Store and load are converted into field elements because they are included into the circuit. Their values are permuted in the sorting network and the resulting witness are used for the memory consistency constraints.